index.js

```diff
@@ -1270,78 +947,15 @@
                    rejectInconsistent);
     };
 
-    //  paramTypesAndValues :: ((a -> Any), Type, [a]) -> ErrorRepr
-    var paramTypesAndValues = function(valFn, type, values) {
-      return map(
-        chain(values, valFn),
-        function(value) {
-          return typesAndValues(
-            type,
-            [value]
-          );
-        }
-      );
-    };
-
-    //  nullaryTypesAndValues :: [Any] -> ErrorRepr
-    var nullaryTypesAndValues = function(values) {
-      return {
-        isParameterizedType: false,
-        types: valueTypes(values),
-        values: values
-      };
-    };
-
-    //  typesAndValues :: (type, [Any]) -> ErrorRepr
-    var typesAndValues = function(t, values) {
-      if (isParameterizedType(t)) {
-        var constructorName = functionName(values[0].constructor);
```

Comment:
> Identity(42).constructor.name;
> // => 'Object'

Could check for `@@type` if constructor is `Object`?

> S.Just(42).constructor.name;
> // => 'Maybe'

The way that Maybe -> Just/Nothing and Either -> Left/Right are defined is pretty unique I would say. It seems pretty odd to have a pseudo base class that is always checking `isJust`, `
Another benefit of this approach is that the error messages will refer to Either#foo rather than Left#foo/Right#foo